gammapy/datasets/utils.py

```diff
@@ -203,3 +205,124 @@ def split_dataset(dataset, width, margin, split_template_models=True):
                 d.models = dataset.models
             datasets.append(d)
     return datasets
+
+
+def create_global_dataset(
+    datasets,
+    name=None,
+    position=None,
+    binsz=None,
+    width=None,
+    energy_min=None,
+    energy_max=None,
+    energy_true_min=None,
+    energy_true_max=None,
+    nbin_per_decade=None,
+):
+    """Create an empty dataset emcompassing the input datasets.
+
+    Parameters
+    ----------
+    datasets : `~gammapy.datasets.Datasets`
+        Datasets
+    name : str, optional
+            Name of the output dataset. Default is None.
+    position : `~astropy.coordinates.SkyCoord`
+        Center position of the output dataset.
+        Defalut is None, and the average postion is taken.
+    binsz : float or tuple or list, optional
+        Map pixel size in degrees.
+        Default is None, the minimum bin size is taken.
+    width : float or tuple or list or string, optional
+        Width of the map in degrees.
+        Default is None, in that case it is derived as
+        the maximal width + twice the maximal separation between datasets and position.
+    energy_min :  `~astropy.units.Quantity`
+        Energy range.
+        Default is None, the minimum energy is taken.
+    energy_max :  `~astropy.units.Quantity`
+        Energy range.
+        Default is None, the maximum energy is taken.
+    energy_true_min :  `~astropy.units.Quantity`
+        Energy range.
+        Default is None, the minimum energy is taken.
+    energy_true_max :  `~astropy.units.Quantity`
+        Energy range.
+        Default is None, the maximum energy is taken.
+    nbin_per_decade : int
+        number of energy bin per decade.
+        Default is None, the maximum is taken.
+    Returns
+    -------
+    datasets : `~gammapy.datasets.MapDatset`
+        Empy global dataset.
+    """
+    from gammapy.datasets import MapDataset
+
+    if position is None:
+        positions = SkyCoord([d.exposure.geom.center_skydir for d in datasets])
+        position = SkyCoord(positions.cartesian.mean(), frame=positions.frame)
+
+    binsz_list = []
```

Comment:
globally isn't it simpler to use list comprehension to build those lists. Is there a case where you want to load the datasets lazily?

Reply:
Not sure it will really make a real difference in performance.